src/v2/user_account.rs

```diff
@@ -0,0 +1,214 @@
+use std::time::SystemTime;
+
+use serde::{Serialize, Deserialize};
+
+use ruma::api::ruma_api;
+use ruma::identifiers::UserId;
+use ruma::api::client::r0::contact::get_contacts::ThirdPartyIdentifier;
+
+use ruma::api::client;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct UserInformation {
+    /// The user's name.
+    pub name: String,
+
+    /// The password hash of th account
+    pub password_hash: String,
+
+    /// Is the account a guest
+    pub is_guest: u64,
+
+    /// Is the user a server admin
+    pub admin: u64,
+
+    // todo: doc but I do not know what this is
+    #[serde(skip_serializing_if="Option::is_none")]
+    pub consent_version: Option<String>,
+
+    // todo: doc but I do not know what this is
+    #[serde(skip_serializing_if="Option::is_none")]
+    pub consent_server_notice_sent: Option<bool>,
+
+    // todo: doc but I do not know what this is
+    #[serde(skip_serializing_if="Option::is_none")]
+    pub appservice_id: Option<String>,
+
+    /// creation date for the account
+    // todo: how to get rid of this option?
+    #[serde(with = "ruma::serde::time::opt_s_since_unix_epoch")]
+    pub creation_ts: Option<SystemTime>,
+
+    // todo: doc but I do not know what this is
+    #[serde(skip_serializing_if="Option::is_none")]
+    pub user_type: Option<String>,
+
+    /// Is the account deactivated
+    pub deactivated: u64,
+
+    /// The user's display name, if set.
+    pub displayname: String,
+
+    /// The user's avatar URL, if set.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub avatar_url: Option<String>,
+
+    /// A list of third party identifiers the homeserver has associated with the user's
+    /// account.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub threepids: Vec<ThirdPartyIdentifier>,
+}
+
+impl Default for UserInformation {
+    fn default() -> Self {
+        UserInformation{
+            creation_ts: Some(SystemTime::now()),
+            ..Default::default()
+        }
+    }
+}
+
+
+pub mod query_user {
```

Comment:
This module should be in its own file, not an inline module declaration.

Reply:
ok, thought I put it into one file cause it is the same endpoint with GET & PUT.
I will separate them

Reply:
How should I continue here?
Regarding the discussion just before I think
`query_user_details::v2` and `create_modify_user::v2` ?
Beside this, should I put all the user admin endpoints into a submodule?

Reply:
I don't care much either way. You can keep putting things in `v2::` and we later sort it out, or you add new things with `v2` being the leaf module, and I'll transition the other ones over soon.

Re. grouping, I think a `users` module would be good, as that is part of the urls too and it seems pointless to repeat `admin` or `api` in the module name when that's part of the crate name already.